### DISCUSSION TITLE
consider requiring formal descriptions for subclasses

### DISCUSSION THREAD

**Discussion by justin2004** [CommonCoreOntology/CommonCoreOntologies#368] [Ontology Terms and Class Schema]
Many subclasses don't formally differentiate themselves from their parent class.

e.g.

`cco:GeopoliticalEntityBorder` is a subclass of `cco:GeospatialBoundary` but `cco:GeopoliticalEntityBorder` does not have a formal definition that distinguishes it from `cco:GeospatialBoundary`. The only thing that distinguishes it from its parent class is human-only readable text.

By formal definition I mean conditions (expressed using some `owl` axioms) that must be met for an instance to be a member of the class.

Here is an example --

Before:
<img width="875" alt="image" src="https://github.com/user-attachments/assets/857bcfe8-ef42-4aa4-9d6b-ebb2fb69f667">
Notice that its Descriptions are the same as its parent class (nothing additional is added).

After:
![image](https://github.com/user-attachments/assets/9333e5da-aa07-4b65-91f5-a0a7fd50993b)
Now it has a Description that adds something to distinguish it from its parent class.

I am not saying that the formation description I added is exactly what `cco:GeopoliticalEntityBorder` needs, thought it might be close, but it would be a better class if it had a formal description that differentiates it from its parent class, right?

**Comment by justin2004** [comment]
another example:

shouldn't `cco:LegalInstrument` be referenced in the definition of `cco:ActOfLegalInstrumentUse`?
that would be a slam dunk.

**Comment by neilotte** [comment]
@justin2004 This is a good observation but it seems to suggest a more sweeping approach to the axiomatization of classes in CCO. For this reason, I'm converting it to a discussion. Please create a separate issue to preserve the comment regarding ActOfLegalInstrument Use, and if you can add a candidate re-phrasing of the definition, that would be very helpful. Thanks!

**Comment by justin2004** [comment]
starting to axiomatise in:
https://github.com/CommonCoreOntology/CommonCoreOntologies/pull/514

**Comment by mark-jensen** [comment]
@justin2004 I like the work you are doing here. This is a hot topic sorely needed to be addressed. I am going cancel your PR for the time being until we can socialize these changes and the overall methodology with a broader audience. Please don't let that hinder your progress in making changes on #514. The work is highly appreciated and won't go to waste. Thanks!

**Reply by justin2004** [reply]
hey @mark-jensen , any update on the socialization of axiomatic class definitions?

**Reply by justin2004** [reply]
hi @mark-jensen , i've got more axiomatic class definitions to add.
i am finding that ontology users in the field prefer to consult the axiomatic class definitions before deciding how to express instance data.
are those going to be welcome in CCO?
